src/sycl/sycl_utils.cpp

```diff
@@ -77,7 +90,11 @@ bool are_equal(const ::sycl::device &lhs, const ::sycl::device &rhs) {
         return gpu::nvidia::compare_cuda_devices(lhs, rhs);
     }
 #endif
-
+#ifdef DNNL_SYCL_HIP
+    if (lhs_be == backend_t::amd) {
+        return gpu::amd::compare_hip_devices(lhs, rhs);
+    }
+#endif
```

Comment:
There is a function below (`sycl_device_id`), that hits assert because AMD backend is not handled there.